### ISSUE TITLE
Undefined or null value gives error when part of an object in an array

### ISSUE THREAD

**Issue by juanreynolds** [JSONPath-Plus/JSONPath#180]
Version 7.2.0 (and earlier)

This code:

```
import { JSONPath } from 'jsonpath-plus';

const obj = {
        arr: [
            {
                type: 'a',
            },
            {
                type: 'b',
                text: undefined as any // THIS IS THE ISSUE
            }
        ]
    }

    let val = JSONPath({
        path: JSONPath.toPathArray('$.arr[?(@.type=="b")].text'), // ISSUE HERE
        json: obj
    });
```
gives error: Error: jsonPath: Cannot read property 'type' of null: @.type=="b"

BUT when the 2nd item in the array has a text property that is not undefined or null it works as expected:

THIS WORKS:

```
const obj = {
        arr: [
            {
                type: 'a',
            },
            {
                type: 'b',
                text: 5   // THIS WORKS
            }
        ]
    }

    let val = JSONPath({
        path: JSONPath.toPathArray('$.arr[?(@.type=="b")].text'),
        json: obj
    });
```

AND when the path is not in array notation but in string notation it works:

THIS WORKS:
```
const obj = {
        arr: [
            {
                type: 'a',
            },
            {
                type: 'b',
                text: undefined as any
            }
        ]
    }

    let val = JSONPath({
        path: '$.arr[?(@.type=="b")].text',  // NOW it works
        json: obj
    });
```

**Comment by mattweberio** [comment]
We have also encountered this bug. In our schema, many values are optional and it never caused a problem with the jsonpath library. However, when using jsonpath-plus, it returns nothing if any item in the array has undefined in the chain (e.g. ".text" is undefined in one object). It would be great if this library could ignore undefined paths and still return the matches.

**Comment by juanreynolds** [comment]
Just tested again and the problem persists in v8.0.0.
The problem only raises its head when working with array notation.

**Comment by 80avin** [comment]
It is fixed in 9.0.0 using ignoreEvalErrors

Check: https://github.com/JSONPath-Plus/JSONPath/issues/218#issuecomment-2383950924
